### PR TITLE
Update use cases

### PR DESCRIPTION
**Pull request by yanhwee**
Refactored default use case (delete contact). Please take a look at the strikethrough text. And the new one below.
Added 2 new use cases (add contact, add notes to contact).

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -314,6 +315,85 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+#### UC3: Add notes to contact
+
+- Precondition: -
+- Postcondition: Notes added to specified contact.
+
+##### MSS
+
+1. User submits contact and new note information.
+2. System checks information is properly formatted.
```
Comment (jialegitgud) [inline docs/DeveloperGuide.md:380]: Can perhaps change to submitted information for better clarity, same for the above use cases that handle user input.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -314,6 +315,85 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+</s>
+
+#### UC1: Delete a contact
+
+- Precondition: -
+- Postcondition: Specified contact is deleted
+
+##### MSS
+
+1. User submits contact information to be deleted.
```
Comment (jialegitgud) [inline docs/DeveloperGuide.md:327]: Perhaps the specific contact information used for deletion can be provided as a guideline

**Review by jialegitgud — COMMENTED** [review]
Generally LGTM, just some small nits with the phrasing.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -314,6 +315,85 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
       Use case resumes at step 2.
 
 *{More to be added}*
+</s>
+
+#### UC1: Delete a contact
+
+- Precondition: -
+- Postcondition: Specified contact is deleted
+
```
Comment (ossaris56) [inline docs/DeveloperGuide.md:324]: Maybe could add User as an actor in too.

**Review by ossaris56 — COMMENTED** [review]
LGTM too, just with a few things missing.

**Pull request merged**